libsolidity/ast/AST.h

```diff
@@ -2625,6 +2625,37 @@ class Builtin: public Expression
 	SourceLocation m_nameParameterLocation;
 };
 
+class ForAllQuantifier: public ASTNode, public Scopable, public ScopeOpener
```

Comment:
One more problem here is NatSpec. Not sure if `ForAllQuantifier` not being `StructurallyDocumented` changes this, but I assume that if you put NatSpec over it, it will not be applied to function. You'd have to put it between `forall` and the function, which would look weird.

Reply:
We don't need to deal with Natspec for this now - the question is rather how do we make sure not to forget about this eventually :-). We should probably start a list of flukes to go over in the future including this... we could then clear out that list regularly (assuming that a lot will change in between and things may become obsolete - but might still be a bit of peace of mind to just *have* such a list to drop stuff like this into)

Reply:
Added a TODO comment for now.

As for the list - do you think an issue a good place for it or would you just have a HackMD note instead?

Reply:
Rather a hackmd note, I'd say

Reply:
[Experimental Solidity quirks to deal with eventually](https://notes.ethereum.org/TLYlbuvqSGKIzRv6F4Mxow).